FirebaseTestProject/Views/PostsList.swift

```diff
@@ -9,10 +9,15 @@ import SwiftUI
 
 struct PostsList: View {
     @StateObject var postData = PostData()
+    @State private var searchText: String = ""
     
     var body: some View {
+        //Searchable Posts
+        SearchBar(text: $searchText)
+        let posts = searchText == "" ? postData.posts : postData.posts.filter({ $0.contains(searchText) })
+        
         NavigationView {
-            List(postData.posts, id: \.text) { post in
+            List(posts, id: \.text) { post in
                 PostRow(post: post)
             }
```

Comment:
Nothing wrong with this per se, but two notes:
* Filtering the posts using an inline variable declaration is a bit unintuitive. I would instead use a conditional statement within the `List` to determine whether to display the `PostRow`.
* I’d prefer checking `searchText.isEmpty` over comparing to an empty string. It’s more readable, and Swift optimizes for it under the hood.

Here’s what I would do instead:

```swift
List(postData.posts, id: \.text) { post in
    if searchText.isEmpty || post.contains(searchText) {
        PostRow(post: post)
    }
}
```